src/dataLoader/savedItemsDataLoader.ts

```diff
@@ -41,10 +41,54 @@ export async function batchGetSavedItemsByUrls(
 }
 
 /**
- * Creates a dataloader for saved items
+ * Reorders savedItems based on the id that are received to
+ * keep the response in the same order of savedItemIds as requested
+ * by the Apollo Gateway (Client API)
+ * @param ids
+ * @param savedItems
+ */
+export function reorderSavedItemsByIds(ids: string[], savedItems: SavedItem[]) {
+  const idToSavedItemMap = savedItems.reduce((acc, savedItem) => {
+    return {
+      ...acc,
+      [savedItem.id]: savedItem,
+    };
+  }, {});
+
+  return ids.map((id) => idToSavedItemMap[id]);
+}
+
+/**
+ * Batch loader function to get savedItems by id.
+ * @param context
+ * @param ids list of savedItem ids.
+ */
+export async function batchGetSavedItemsByIds(
+  context: IContext,
+  ids: string[]
+) {
+  const savedItems = await new SavedItemDataService(
+    context
+  ).batchGetSavedItemsByGivenIds(ids);
+
+  return reorderSavedItemsByIds(ids, savedItems);
+}
+
+/**
+ * Creates a dataloader for saved items to fetch by id
+ * @param context
+ */
+export function createSavedItemsDataLoaderById(context: IContext) {
```

Comment:
won't block on this comment, but since there are two keys for loading savedItem now it could be useful to fill both caches when loading.

See [documentation](https://github.com/graphql/dataloader#loading-by-alternative-keys) here

Reply:
i tried implementing this, but coz our `batchGetSavedItemsByIds` requires `context` as one of the params and with `DataLoader` accepting only id/url and not context, this gets tricky. might have to decouple the usage of context from `batchGetSavedItemsByIds`

in the below code, `prime` doesn't apply to `savedItemByUrlDataLoader` as its a function and not DataLoader object
```
const savedItemByIdDataLoader = (context: IContext, ids: string[]) => {
  let dl = new DataLoader(async (ids: string[]) => {
    let savedItems = await batchGetSavedItemsByIds(context, ids);
    for (let item of savedItems) {
      savedItemByUrlDataLoader.prime(item.id, item);
    }
    return savedItems;
  });
  return dl;
};
```

Reply:
Ohh, if we're not sharing dataloaders across resolvers then I don't think we're actually using dataloader cache

Reply:
Let's not make any changes, we can update as needed in another ticket.